packages/core/test/errors.test.ts

```diff
@@ -892,4 +893,41 @@ describe('error handling', () => {
       error_thrown_in_guard_when_transitioning]
     `);
   });
+
+  it('error thrown when resolving the initial context should rethrow synchronously', (done) => {
+    const machine = createMachine({
+      context: () => {
+        throw new Error('oh no');
+      }
+    });
+
+    const actor = createActor(machine);
+
+    installGlobalOnErrorHandler(() => {
+      done.fail();
+    });
+
+    expect(() => actor.start()).toThrowErrorMatchingInlineSnapshot(`"oh no"`);
+
+    setTimeout(() => {
+      done();
+    }, 10);
+  });
+
+  it('error thrown when resolving the initial context should reject when wrapped in a Promise', async () => {
+    const machine = createMachine({
+      context: () => {
+        throw new Error('oh no');
+      }
+    });
+
+    const actor = createActor(machine);
+
+    try {
+      await toPromise(actor.start());
+      fail();
+    } catch (err) {
+      expect((err as Error).message).toEqual('oh no');
+    }
+  });
```

Comment:
This doesn't work like you think it does. This error is still very much thrown synchronously. `toPromise` isn't ever called during this test case.

Reply:
I'm not sure what you mean.  If `createActor` threw synchronously, wouldn't this test fail?